src/Api.js

```diff
@@ -89,13 +92,68 @@ class Api {
 	}
 
 	/**
-	 * Get the WhoColor data for a given wiki page.
+	 * Get the WikiWho data for a given wiki page. Use onError and onResults to set return handlers.
 	 *
-	 * @param  {string} wikiUrl URL of the wiki page that we want to analyze.
-	 * @return {Promise}
+	 * @param {string} wikiUrl URL of the wiki page that we want to analyze.
 	 */
 	getData( wikiUrl ) {
-		return $.getJSON( this.getAjaxURL( wikiUrl ) );
+		$.getJSON( this.getAjaxURL( wikiUrl ) ).then( ( function ( result ) {
+			// Handle error.
+			if ( !result.success && result.info && this.errorHandler ) {
+				// Turn the English error message into a i18n message key.
+				// @TODO Remove this when the WikiWho API returns useful error codes.
+				let errorCode = result.info
+					.toLowerCase()
+					.replace( / /g, '-' )
+					.replace( /[^a-z-]/g, '' );
+				// Possible messages:
+				// eslint-disable-next-line
+				// ext-whowrotethat-requested-data-is-not-currently-available-in-wikiwho-database-it-will-be-available-soon
+				this.errorHandler( 'ext-whowrotethat-' + errorCode );
+				return;
+			}
+
+			let cleanedResults = {
```

Comment:
This could also be a `const` to prevent re-declaration (insignificant here, but can help prevent bugs in larger applications)

Reply:
Code removed.